src/StructTable.cpp

```diff
@@ -10,6 +10,23 @@
 
 namespace cmm
 {
+    StructData::StructData(const EnumSymState symState, std::unordered_map<std::string, Field>&& fieldMap) :
+        symState(symState), fieldMap(std::move(fieldMap))
+    {
+    }
+
+    Field* StructData::findField(const std::string& name) CMM_NOEXCEPT
```

Comment:
Consider using the IField interface.

Reply:
After reviewing the code, I decided to keep as-is to avoid needing to do extra heap allocation.  Any 'getter' type functions will instead use IField, but keep this storage as a "normal" Field.